src/components/study-manager.js

```diff
@@ -89,7 +92,9 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const StudyCard = ({study, onClick}) => {
-    const [open, setOpen] = React.useState(false);
+    const [openDeleteDialog, setOpenDelete] = React.useState(false);
+    const [openRenameDialog, setOpenRename]  = React.useState(false);
+    const [newStudyNameValue, setNewStudyName] = React.useState(study.studyName);
```

Comment:
I think it's weird to put this state in the studycard. Maybe extract it to another component ? The alternative is to use uncontrolled forms but that's a bit of an antipattern in react

Reply:
To be more precise, can we use more components to restrict the scope of relayouting (at each keystroke in input in the dialog) ?

ie move the studyName in something like:
```
const renameDialog = () => {
     const [newStudyNameValue, setNewStudyName] = React.useState(study.studyName); 
}
```

Reply:
@jonenst I made the change for renameDialog, is that what you expected ?